src/sentry/receivers/outbox/control.py

```diff
@@ -67,23 +76,30 @@ def process_async_webhooks(payload: Mapping[str, Any], region_name: str, **kwds:
     webhook_payload = ControlOutbox.get_webhook_payload_from_outbox(payload=payload)
 
     if SiloMode.get_current_mode() == SiloMode.CONTROL:
-        # By default, these clients will raise errors on non-20x response codes
-        response = RegionSiloClient(region=region).request(
-            method=webhook_payload.method,
-            path=webhook_payload.path,
-            headers=webhook_payload.headers,
-            # We need to send the body as raw bytes to avoid interfering with webhook signatures
-            data=webhook_payload.body,
-            json=False,
-        )
-        logger.info(
-            "webhook_proxy.complete",
-            extra={
-                "status": response.status_code,
-                "request_path": webhook_payload.path,
-                "request_method": webhook_payload.method,
-            },
-        )
+        try:
+            hash_id = hash((region_name, shard_identifier, object_identifier))
+            retry_deadline = timedelta(minutes=5)
+            # By default, these clients will raise errors on non-2xx response codes
+            client = RegionSiloClient(region=region, retry_deadline=retry_deadline, hash_id=hash_id)
+            response = client.request(
+                method=webhook_payload.method,
+                path=webhook_payload.path,
+                headers=webhook_payload.headers,
+                # We need to send the body as raw bytes to avoid interfering with webhook signatures
+                data=webhook_payload.body,
+                json=False,
+            )
+
+            logger.info(
+                "webhook_proxy.complete",
+                extra={
+                    "status": response.status_code,
+                    "request_path": webhook_payload.path,
+                    "request_method": webhook_payload.method,
+                },
+            )
+        except SiloClientError as e:
```

Comment:
This was a bug where webhook outbox messages would've been retried on `SiloClientError`. I've caught them here and generated a Sentry error event.